### PR TITLE
Add Method/Ctor source code to JavaProject

### LINKED ISSUE #89: Attach Java source code to methods and constructors

### PR DESCRIPTION
**Pull request by john-zou**
image

closes #89

### CODE REVIEW

**Comment by john-zou** [comment]
It seems the test is failing due to line endings, I'll modify the tests to not look at line endings.

image

File: src/main/java/com/github/algobot76/surabaya/util/Analyzer.java [hunk 1]
```diff
@@ -30,18 +40,29 @@ public Analyzer() {
...
+						CompilationUnit cu = (CompilationUnit) result.getResult().get();
+						String[] lines = content.lines().toArray(String[]::new);
+						staticAnalysisVisitor.setFileSrc(lines);
+						staticAnalysisVisitor.visit(cu, project);
+					}
+					catch (IOException e) {
+						logger.error(e.getMessage());
+					}
+				}
+			});
```
Comment (algobot76) [inline src/main/java/com/github/algobot76/surabaya/util/Analyzer.java:60]: This is unnecessary as `SourceRoot` will handle the file reading properly for us.

File: src/main/java/com/github/algobot76/surabaya/util/StaticAnalysisVisitor.java [hunk 2]
```diff
@@ -98,8 +110,13 @@ public void visit(MethodDeclaration n, Project project) {
...
 	@Override
 	public void visit(ConstructorDeclaration n, Project project) {
+		if (n.getRange().isEmpty()) {
+			return;
+		}
+		Range range = n.getRange().get();
+		String src = getSrcFromRange(range);
 		Constructor constructor = new Constructor(n.getName().getIdentifier().trim(),
-				getAccessModifier(n.getModifiers()));
+				getAccessModifier(n.getModifiers()), src);
```
Comment (algobot76) [inline src/main/java/com/github/algobot76/surabaya/util/StaticAnalysisVisitor.java:119]: The easiest way to get the code of the constructor is to call the `toString` method on `n`.

File: src/main/java/com/github/algobot76/surabaya/util/StaticAnalysisVisitor.java [hunk 3]
```diff
@@ -108,6 +125,34 @@ public void visit(ConstructorDeclaration n, Project project) {
...
+		String lastLineStr = this.currentFileSrc[lastLine];
+
+		// https://stackoverflow.com/questions/52559564/call-to-substring-is-redundant
+		// This does the same as lastLineStr.substring(0, range.end.column);
+		sb.append(lastLineStr, 0, range.end.column);
+		sb.append(System.lineSeparator());
+
+		return sb.toString();
+	}
+
```
Comment (algobot76) [inline src/main/java/com/github/algobot76/surabaya/util/StaticAnalysisVisitor.java:155]: We could simply use `toString` method to get the source code of a method/constructor. This helper method seems unnecessary.

File: src/main/java/com/github/algobot76/surabaya/util/Analyzer.java (hunk 1 above)
Comment (john-zou) [reply src/main/java/com/github/algobot76/surabaya/util/Analyzer.java:60]: SourceRoot doesn't give access to original code, that's why I switched to JavaParsing each file alongside reading flie contents with Files.readString

**Comment by john-zou** [comment]
Ok thanks @algobot76 , made the changes, this is much better than the original, it formats the original code too, which is nice

**Comment by john-zou** [comment]
Included in #102

**Pull request closed without merging**